diffrax/term.py

```diff
@@ -422,8 +424,8 @@ def is_vf_expensive(
         y: Tuple[PyTree, PyTree, PyTree, PyTree],
         args: PyTree,
     ) -> bool:
-        control = self.contr(t0, t1)
-        if sum(c.size for c in jtu.tree_leaves(control)) in (0, 1):
+        control_struct = jax.eval_shape(self.contr, t0, t1)
```

Comment:
This is just a drive-by improvement, to avoid the evaluation in the first place.